polars/polars-core/src/series/mod.rs

```diff
@@ -627,6 +627,16 @@ impl Series {
         }
     }
 
+    /// Count the number of elements elements over an expanding window.
+    pub fn cumcount(&self, reverse: bool, count_nulls: bool) -> Series {
```

Comment:
This is much more expensive than the old implemenation. You can check if `null_count==0` and then use the old implementation for this as that was simply a cheap arange.

Reply:
I took an entirely new approach in my latest push -- it should be much more efficient now